tests/tasks/aws/test_step_function.py

```diff
@@ -4,31 +4,29 @@
 
 pytest.importorskip("boto3")
 
-import prefect
 from prefect.tasks.aws import StepActivate
-from prefect.utilities.configuration import set_temporary_config
 
 
 class TestStepActivate:
     def test_initialization(self):
-        task = StepActivate(state_machine_arn="arn", execution_name="name")
+        task = StepActivate(state_machine_arn="arn")
 
     def test_initialization_passes_to_task_constructor(self):
-        task = StepActivate(
-            state_machine_arn="arn", execution_name="name", name="test", tags=["AWS"]
-        )
+        task = StepActivate(state_machine_arn="arn", name="test", tags=["AWS"])
         assert task.name == "test"
         assert task.tags == {"AWS"}
 
     def test_exposes_boto3_start_execution_api(self, monkeypatch):
-        task = StepActivate(state_machine_arn="arn", execution_name="name")
+        task = StepActivate(state_machine_arn="arn")
         client = MagicMock()
         boto3 = MagicMock(client=client)
         monkeypatch.setattr("prefect.utilities.aws.boto3", boto3)
-        task.run()
+        task.run(execution_name="name")
 
         called_method = client.mock_calls[1]
         assert called_method[0] == "().start_execution"
-        called_method.assert_called_once_with(
-            {"stateMachineArn": "arn", "name": "name", "input": {}}
-        )
+        assert called_method[2] == {
+            "stateMachineArn": "arn",
+            "name": "name",
+            "input": "{}",
+        }
```

Comment:
This can just be 
@mmiermans — I'm not sure what was going on with the assertion before, but this is the proper way to use these mocks:
```python
        client().start_execution.assert_called_once_with(
            stateMachineArn="arn", name="name", input="{}"
        )
```

If you add `x=1` or something to this call, the assertion will fail as intended.

Reply:
Thank you, that's it!

For my understanding, could you explain why it should be `client().start_execution` and not `client.start_execution`?

I'll apply the same fix to other tests in a future PR.

Reply:
Sorry, missed this comment :)

`client.start_execution` would assert that the class method `Client.start_execution` was called, but we want to assert that the instance method `Client().start_execution` was called. It could also be written as `client.return_value.start_execution.assert...`

```python
class Foo:
    def hello(self):
        print("Hello")

from unittest.mock import MagicMock

mock = MagicMock()
Foo = mock  # pretend to do a monkeypatch

instance = Foo()
instance.hello()

mock.hello.assert_not_called()
mock().hello.assert_called_once()
```